### PR TITLE
[PDI-17297] Filters Exclude and Keep Only, don't work properly with D…

### PR DESCRIPTION
**Pull request by ssamora**
…ate type
@pentaho-lmartins @pentaho/kashyyyk 

This is part two of two pull requests.
The other is:
https://github.com/pentaho/pentaho-det-ee/pull/555

These two pull requests for issue PDI-17297 also solves PDI-17296

### CODE REVIEW

**Comment by joaormf** [comment]
@pentaho/bobafett please review also

File: impl/src/main/java/org/pentaho/di/trans/dataservice/optimization/ValueMetaResolver.java [hunk 1]
```diff
@@ -66,15 +66,21 @@ public ValueMetaResolver( RowMetaInterface rowMeta ) {
 
         switch ( valueMetaInterface.getType() ) {
           case ValueMetaInterface.TYPE_DATE:
-            valueMetaInterface.setConversionMask( ANSI_DATE_LITERAL );
+            if ( valueMetaInterface.getConversionMask() == null ) {
```
Comment (joaormf) [inline impl/src/main/java/org/pentaho/di/trans/dataservice/optimization/ValueMetaResolver.java]: Shouldn't we check for null or empty string?

Comment (ssamora) [reply impl/src/main/java/org/pentaho/di/trans/dataservice/optimization/ValueMetaResolver.java]: @joaormf Yes, agree with that, if the mask is empty we will fall in the same error. I will update this.

**Comment by ssamora** [comment]
PR was updated

**Pull request merged**